jleeboyd/family-bank-app
# TRANSACTION: Delete Transactions

### User Story
Resolves issues: 
#44 - Delete transaction button
#34 - Submitting empty form crashed app

This PR creates functionality to delete transactions from the transaction history. When a user clicks the delete button, the most recently created transaction will be crossed out and the account balance will be reverted by that amount.

This PR also fixes the issue where the app would crash if no form data was entered on account creation.


Account dialog toast:
![Screen Shot 2020-11-11 at 9 19 00 PM](https://user-images.githubusercontent.com/52220118/98898990-a267b580-2463-11eb-8ad8-13334177d866.png)

Transaction dialog toast:
![Screen Shot 2020-11-11 at 9 19 22 PM](https://user-images.githubusercontent.com/52220118/98899020-aeec0e00-2463-11eb-9b60-14e5b96cb89f.png)

Transaction deletion:
![Screen Shot 2020-11-11 at 9 19 30 PM](https://user-images.githubusercontent.com/52220118/98899035-b7dcdf80-2463-11eb-8fbc-e22c539f89bb.png)


### Context / Additional Information
The button does not include a secondary confirm currently.

To Test:
1. Create an account and enter no information, then click confirm. See that the app does not crash, and a toast message appears.

2. Create an account with a non zero balance
3. Click the account
4. Make a withdraw or deposit with no form information. See that a toast message appears

5. Now make a withdraw or deposit with a nonzero value
6. Click the delete button
7. See that the transaction is crossed out, and the balance has reverted by that same amount.

### Acceptance Criteria
- [x] - Delete button crosses out the last valid transaction
- [x] - Toasts when there are no more valid transactions
- [x] - App does not crash when empty form is submitted, and displays toast
